core/src/main/scala/kafka/tools/ConsumerPerformance.scala

```diff
@@ -136,11 +136,12 @@ object ConsumerPerformance {
           bytesRead += record.key.size
         if(record.value != null)
           bytesRead += record.value.size 
-      
-        if (messagesRead % config.reportingInterval == 0) {
+
+        val currentTimeMillis = System.currentTimeMillis
```

Comment:
Sure and a nice catch. One option is to make `currentTimeMillis` a variable and initially assign it before the while loop at line 129, and then re-assign the current time to it inside the while loop (to replace the current `val` definition at line 140). I'll update the PR.